.github/workflows/test.yml

```diff
@@ -0,0 +1,150 @@
+name: Test Suite
+
+on:
+  push:
+    branches:
+      - staging
+  pull_request:
+
+jobs:
+  unity-testsuite:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      # Grab the SpacetimeDB branch name from the PR description. If it's not found, master will be used instead.
+      # We'll use this branch name for any integration tests with SpacetimeDB.
+
+      - name: Extract SpacetimeDB branch name or PR link from PR description
+        id: extract-branch
+        if: github.event_name == 'pull_request'
+        env:
+          description: ${{ github.event.pull_request.body }}
+        run: |
+          # Check if description contains a branch name or a PR link
+          branch_or_pr=$(echo "$description" | grep -oP '(?<=SpacetimeDB branch name:\s).+')
+          echo "Branch or PR found: $branch_or_pr"
+
+          if [[ -z "$branch_or_pr" ]]; then
+            branch="master"
+          elif [[ "$branch_or_pr" =~ ^https://github.com/.*/pull/[0-9]+$ ]]; then
+            # If it's a PR link, extract the branch name from the PR
+            pr_number=$(echo "$branch_or_pr" | grep -oP '[0-9]+$')
+            branch=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+            https://api.github.com/repos/clockworklabs/SpacetimeDB/pulls/$pr_number | jq -r '.head.ref')
+          else
+            # It's already a branch name
+            branch="$branch_or_pr"
+          fi
+
+          echo "branch=$branch" >> $GITHUB_OUTPUT
+          echo "Final branch name: $branch"
+
+      - name: Checkout SpacetimeDB
+        uses: actions/checkout@v4
+        with:
+          repository: clockworklabs/SpacetimeDB
+          ref: ${{ steps.extract-branch.outputs.branch }}
+          path: SpacetimeDB~
+
+      # Run cheap .NET tests first. If those fail, no need to run expensive Unity tests.
+
+      - name: Override NuGet packages
+        run: |
+          dotnet pack SpacetimeDB~/crates/bindings-csharp/BSATN.Runtime
+
+          # Write out the nuget config file to `nuget.config`. This causes the spacetimedb-csharp-sdk repository
+          # to be aware of the local versions of the `bindings-csharp` packages in SpacetimeDB, and use them if
+          # available. Otherwise, `spacetimedb-csharp-sdk` will use the NuGet versions of the packages.
+          # This means that (if version numbers match) we will test the local versions of the C# packages, even
+          # if they're not pushed to NuGet.
+          # See https://learn.microsoft.com/en-us/nuget/reference/nuget-config-file for more info on the config file.
+          ./tools~/write-nuget-config.sh SpacetimeDB~
+
+      - name: Run .NET tests
+        run: dotnet test -warnaserror
```

Comment:
How long to the dotnet tests take to run? In principle, this runs something in serial that was previously run in parallel, although if this doesn't take much time it doesn't really matter.

Reply:
They are very quick compared to Unity tests, like less than 10 sec or so, so I figured it's fine.